MonoGame.Framework/Graphics/Effect/Effect.cs

```diff
@@ -228,68 +191,120 @@ internal protected override void GraphicsDeviceResetting()
 
         #region Effect File Reader
 
-		private void ReadEffect (BinaryReaderEx reader)
-		{
-			// TODO: Maybe we should be reading in a string 
-			// table here to save some bytes in the file.
-
-			// Read in all the constant buffers.
-			var buffers = (int)reader.ReadByte ();
-			ConstantBuffers = new ConstantBuffer[buffers];
-			for (var c = 0; c < buffers; c++) 
-            {				
-				var name = reader.ReadString ();               
-
-				// Create the backing system memory buffer.
-				var sizeInBytes = (int)reader.ReadInt16 ();
-
-				// Read the parameter index values.
-				var parameters = new int[reader.ReadByte ()];
-				var offsets = new int[parameters.Length];
-				for (var i = 0; i < parameters.Length; i++) 
-                {
-					parameters [i] = (int)reader.ReadByte ();
-					offsets [i] = (int)reader.ReadUInt16 ();
-				}
+        struct MGFXHeader
+        {
+            /// <summary>
+            /// The MonoGame Effect file format header identifier ("MGFX"). 
+            /// </summary>
+            public static readonly int MGFXSignature = (BitConverter.IsLittleEndian) ? 0x5846474D : 0x4D474658;
 
-                var buffer = new ConstantBuffer(GraphicsDevice,
-				                                sizeInBytes,
-				                                parameters,
-				                                offsets,
-				                                name);
-                ConstantBuffers[c] = buffer;
+            /// <summary>
+            /// The current MonoGame Effect file format version.            
+            /// </summary>            
+            public const int MGFXVersion = 10;
+
+            /// <summary>
+            /// The minimum MonoGame Effect file format version supported (inclusive).
+            /// </summary>
+            /// <remarks>
+            /// We should avoid supporting old versions for very long if at all 
+            /// as users should be rebuilding content when packaging their game.
+            /// </remarks>
+            public const int MGFXMinVersion = 9;
```

Comment:
We never support older generated versions of shaders as there is no good reason to.

Reply:
Scratch this, Tom wants it.

Reply:
Okey

Reply:
@harry-cpp I've just updated the PR, version 9 compatibility removed. Since the older shaders no longer work I had to regenerate both the core shaders and the ones used in the content effect tests.

I've ran all the tests and everything seems good, you might want to try regeneraring the MGFXO files on your own for extra safety.